Lib/test/test_json/test_encode_basestring.py

```diff
@@ -0,0 +1,54 @@
+from test.test_json import PyTest, CTest
+from test.support import bigaddrspacetest
+
+
+CASES = [
+    (
+        '/\\"\ucafe\ubabe\uab98\ufcde\ubcda\uef4a\x08\x0c\n\r\t`1~!@#$%^&*()_+-=[]{}|;:\',./<>?',
+        '"/\\\\\\"\ucafe\ubabe\uab98\ufcde\ubcda\uef4a\\b\\f\\n\\r\\t`1~!@#$%^&*()_+-=[]{}|;:\',./<>?"',
+    ),
+    (
+        '\u0123\u4567\u89ab\ucdef\uabcd\uef4a',
+        '"\u0123\u4567\u89ab\ucdef\uabcd\uef4a"',
+    ),
+    ('controls', '"controls"'),
+    ('\x08\x0c\n\r\t', '"\\b\\f\\n\\r\\t"'),
+    (
+        '{"object with 1 member":["array with 1 element"]}',
+        '"{\\"object with 1 member\\":[\\"array with 1 element\\"]}"',
+    ),
+    (' s p a c e d ', '" s p a c e d "'),
+    ('\U0001d120', '"\U0001d120"'),
+    ('\u03b1\u03a9', '"\u03b1\u03a9"'),
+    ("`1~!@#$%^&*()_+-={':[,]}|;.</>?", '"`1~!@#$%^&*()_+-={\':[,]}|;.</>?"'),
+    ('\x08\x0c\n\r\t', '"\\b\\f\\n\\r\\t"'),
+    (
+        '\u0123\u4567\u89ab\ucdef\uabcd\uef4a',
+        '"\u0123\u4567\u89ab\ucdef\uabcd\uef4a"',
+    ),
+]
+
+
+class TestEncodeBasestring:
```

Comment:
`encode_basestring` is not a part of the public API, it doesn't need direct tests.

Reply:
I've replaced it with a test of `dumps`